cmd/kubeadm/app/phases/controlplane/users_linux.go

```diff
@@ -0,0 +1,260 @@
+// +build linux
+
+package controlplane
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"strconv"
+	"strings"
+
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/utils/exec"
+)
+
+var ()
+
+var UpdateFileOwnershipFn = func(filepath string, uid int64, gid int64, permissions uint32) error {
+	if err := os.Chown(filepath, int(uid), int(gid)); err != nil {
+		return err
+	}
+	if err := os.Chmod(filepath, fs.FileMode(permissions)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func UpdateFileOwnership(filepath string, uid int64, gid int64, permissions uint32) error {
+	return UpdateFileOwnershipFn(filepath, uid, gid, permissions)
+}
+
+var UpdateDirecotryOwnershipFn = func(directorypath string, uid int64, gid int64, permissions string) error {
```

Comment:
i see that this is used for etcd.
kubeadm pre-creates the etcd directory with some permissions and then leaves it be.

why do we need to do this recursively at startup?
at cluster creation the directory is mostly empty, but then etcd takes over starts writing more snapshot related files.
can't we just chown the parent etcd directory but not contents?